website/blog/2024-08-07-manipulate-devtools.md

```diff
@@ -0,0 +1,325 @@
+---
+title: "Interacting with Chrome DevTools from WebDriverIO"
+authors: George Romanas & Nikos Lytras
+authorsImageURL: https://avatars.githubusercontent.com/u/48865319?s=400&u=e292306d956ecd6f5d94ff8cde2bc189841c270a&v=4
+                 https://avatars.githubusercontent.com/u/150701199?v=4
+---
+
+# Interacting with Chrome DevTools from WebDriverIO: A Guide
+
+One of WebDriverIO features is the ability to interact with Chrome DevTools, which can be extremely helpful for debugging and testing web applications and browser extensions. This guide will show you how to perform any interaction with DevTools using WebDriverIO v8 and will follow the WebDriverIO v7 guide. In this blog, we'll explore how to use WebDriverIO to interact with the Chrome DevTools, specifically focusing on interacting with the PixiJS extension.
+You can find a complete example in [Github](https://github.com/gromanas/wdio-devtools-demo)
+
+# Devtools in WebdriverIO v8
+
+## Setting Up Extension
+
+You need to download the extension you want to test in a `.crx` format,
+in our case we have downloaded the `PixiJS-Devtools-Chrome-Web-Store.crx` extension file.
+
+## Setting Up WebDriverIO
+
+First, ensure you have WebDriverIO set up in your project. If not, you can install it by following the official installation guide [here](https://webdriver.io/docs/gettingstarted#initiate-a-webdriverio-setup)
+
+## WebDriverIO Configuration for Chrome DevTools
+
+Here is the configuration file (`wdio.conf.js`) tailored to our example:
+
+```javascript
+import path from "path";
+import fs from "fs";
+
+export const config = {
+  specs: ["./test/specs/**/*.js"],
+  runner: "local",
+  maxInstances: 1,
+  capabilities: [
+    {
+      browserName: "chrome",
+      browserVersion: "stable",
+      "goog:chromeOptions": {
+        args: [
+          "ignore-certificate-errors-spki-list",
+          "--ignore-certificate-errors",
+          "window-size=1920,1080",
+          "--auto-open-devtools-for-tabs",
+        ],
+        extensions: [
+          fs.readFileSync(
+            path.resolve("./PixiJS-Devtools-Chrome-Web-Store.crx"),
+            "base64"
+          ),
+        ],
+      },
+      "goog:loggingPrefs": {
+        browser: "INFO",
+      },
```

Comment:
Are these options required for the example? The less configuration the better here I would say

Reply:
done